tests/API/Management/ClientsTest.php

```diff
@@ -2,42 +2,197 @@
 namespace Auth0\Tests\API\Management;
 
 use Auth0\SDK\API\Management;
+use Auth0\SDK\API\Management\Clients;
 use Auth0\Tests\API\BasicCrudTest;
 
-class ClientsTest extends BasicCrudTest {
-    
-    protected function getId($entity) {
-        return $entity['client_id'];
-    }
+/**
+ * Class ClientsTest
+ *
+ * @package Auth0\Tests\API\Management
+ */
+class ClientsTest extends BasicCrudTest
+{
+    /**
+     * Unique identifier name for Clients.
+     *
+     * @var string
+     */
+    protected $id_name = 'client_id';
+
+    /**
+     * Name of the created Client.
+     * Appended with a random string in self::__construct().
+     *
+     * @var string
+     */
+    protected $create_client_name = 'TEST-CREATE-CLIENT-';
+
+    /**
+     * Application type of the created Client.
+     *
+     * @var string
+     */
+    protected $create_app_type = Clients::APP_TYPE_REGULAR_WEB;
+
+    /**
+     * SSO setting of the created Client.
+     *
+     * @var bool
+     */
+    protected $create_sso = false;
+
+    /**
+     * Description of the created Client.
+     *
+     * @var string
+     */
+    protected $create_desc = '__Auth0_PHP_initial_app_description__';
 
-    protected function getApiClient() {
-        $env = $this->getEnv();
-        $token = $this->getToken($env, [
-            'clients' => [
-                'actions' => ['create', 'read', 'delete', 'update']
-            ]
-        ]);
+    /**
+     * Name of the updated Client.
+     * Appended with a random string in self::__construct().
+     *
+     * @var string
+     */
+    protected $update_client_name = 'TEST-UPDATE-CLIENT-';
 
-        $this->domain = $env['DOMAIN'];
+    /**
+     * Application type of the updated Client.
+     *
+     * @var string
+     */
+    protected $update_app_type = Clients::APP_TYPE_NON_INTERACTIVE;
 
-        $api = new Management($token, $env['DOMAIN']);
+    /**
+     * SSO setting of the updated Client.
+     *
+     * @var bool
+     */
+    protected $update_sso = true;
 
+    /**
+     * Description of the updated Client.
+     *
+     * @var string
+     */
+    protected $update_desc = '__Auth0_PHP_updated_app_description__';
+
+    /**
+     * ClientsTest constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->create_client_name .= rand();
+        $this->update_client_name .= rand();
+    }
+
+    /**
+     * Return the ID for the entity created.
+     *
+     * @param array $entity - Client created during test.
+     *
+     * @return string|integer
+     */
+    protected function getId($entity)
+    {
+        return $entity[$this->id_name];
+    }
+
+    /**
+     * Return the Clients API to test.
+     *
+     * @return Management\Clients
+     */
+    protected function getApiClient()
+    {
+        $token = $this->getToken($this->env, [ 'clients' => [ 'actions' => ['create', 'read', 'delete', 'update' ] ] ]);
```

Comment:
IMHO you shouldn't be running tests against the real server here in a public repo. I don't know how you're obtaining the token since I'm not digging in that method, but this should probably be part of a separate module and executed with the right credentials through something like jenkins.
Apart from that, this call is async since is requesting a token, right? Is it OK to use immediately the token right below?